pulp_smash/tests/rpm/api_v2/test_sync_publish.py

```diff
@@ -333,3 +335,76 @@ def get_feed(self, repo):
         """Build the feed to an RPM repository's distributor."""
         feed = urljoin(self.cfg.base_url, 'pulp/repos/')
         return urljoin(feed, repo['distributors'][0]['config']['relative_url'])
+
+
+class SyncInParallelTestCase(unittest.TestCase):
+    """Sync several repositories in parallel."""
+
+    def test_all(self):
+        """Sync several repositories in parallel.
+
+        Specifically, do the following:
+
+        1. Create several repositories. Ensure each repository has an importer
+           whose feed references a repository containing one or more errata.
+        2. Sync each repository. Assert each sync completed successfully.
+        3. Get a summary of information about each repository, and assert the
+           repo has an appropriate number of errata.
+
+        `Pulp #2721`_ describes how a race condition can occur when multiple
+        repos with identical errata are synced at the same time. This test case
+        attempts to trigger that race condition.
+
+        .. _Pulp #2721: https://pulp.plan.io/issues/2721
+        """
+        cfg = config.get_config()
+        client = api.Client(cfg, api.json_handler)
+        repos = []  # append() and pop() are thread-safe
+
+        def create_repo():
+            """Create a repository and schedule its deletion.
+
+            Append a dict of information about the repository to ``repos``.
+            """
+            body = gen_repo()
+            body['importer_config']['feed'] = RPM_UNSIGNED_FEED_URL
+            repo = client.post(REPOSITORY_PATH, body)
+            self.addCleanup(client.delete, repo['_href'])
+            repos.append(repo)
+
+        def get_repo(repo):
+            """Get information about a repository. Append it to ``repos``.
+
+            :param repo: A dict of information about a repository.
+            """
+            repos.append(client.get(repo['_href']))
+
+        threads = tuple(Thread(target=create_repo) for _ in range(5))
+        for thread in threads:
+            thread.start()
+        for thread in threads:
+            thread.join()
```

Comment:
I don't know either, what I know is that multiprocessing is able to return values so maybe using it can make the code shorter, but not that shorter so I think using the current approach is fine.